### ISSUE TITLE
Incorrect semantics for "give"

### ISSUE THREAD

**Issue by gabbard** [isi-vista/adam#693]
At step 250 of the phase 1 curriculum, we think "give" implies that the one person is in contact with the given object at `AFTER` but the other has a `HAS` relationship to it at the same time.

**Comment by gabbard** [comment]
[slot1_gives_slot2_slot3.pdf](https://github.com/isi-vista/adam/files/4371800/slot1_gives_slot2_slot3.pdf)

**Comment by gabbard** [comment]
Another problem: SLOT1 has an `exterior-but-in-contact` relation at `AFTER` with SLOT2.

**Comment by gabbard** [comment]
Here is the relevant `ActionDescription`:

```
_GIVE_ACTION_DESCRIPTION = ActionDescription(
    frame=ActionDescriptionFrame(
        {AGENT: _GIVE_AGENT, THEME: _GIVE_THEME, GOAL: _GIVE_GOAL}
    ),
    enduring_conditions=[
        bigger_than(_GIVE_AGENT, _GIVE_THEME),
        bigger_than(_GIVE_GOAL, _GIVE_THEME),
        # partOf(_GIVE_AGENT_MANIPULATOR, _GIVE_AGENT),
        # partOf(_GIVE_GOAL_MANIPULATOR, _GIVE_GOAL),
    ],
    preconditions=[
        has(_GIVE_AGENT, _GIVE_THEME),
        negate(has(_GIVE_GOAL, _GIVE_THEME)),
        contacts(_GIVE_AGENT_MANIPULATOR, _GIVE_THEME),
        negate(contacts(_GIVE_GOAL_MANIPULATOR, _GIVE_THEME)),
    ],
    postconditions=[
        negate(has(_GIVE_AGENT, _GIVE_THEME)),
        has(_GIVE_GOAL, _GIVE_THEME),
        negate(contacts(_GIVE_AGENT_MANIPULATOR, _GIVE_THEME)),
        contacts(_GIVE_GOAL_MANIPULATOR, _GIVE_THEME),
    ],
    asserted_properties=[
        (_GIVE_AGENT, VOLITIONALLY_INVOLVED),
        (_GIVE_AGENT, CAUSES_CHANGE),
        (_GIVE_THEME, UNDERGOES_CHANGE),
    ],
)
```

It is possible that `negate` is not working correctly.

**Comment by gabbard** [comment]
Closed by #709
